### PR TITLE
[FIX] #6642 Strip Markdown from conversation title

### PR DESCRIPTION
**Pull request by SaiVardhan**
Strip Markdown from conversation title

### CODE REVIEW

**Comment by jhass** [comment]
Tests :)

`Message` already has `#message` which returns a `MessageRenderer` for `#text`. `Conversation` should probably gain a similar convenience method for returning a wrapped `#subject`.

Though I'm not entirely sure we should render markdown there, especially in the subject, it's rendering links inside links, isn't it? Perhaps `MessageRenderer#title` is more appropriate?

**Comment by Flaburgan** [comment]
>Though I'm not entirely sure we should render markdown there

**Comment by SuperTux88** [comment]
Why is there markdown in the subject anyway? this field doesn't support markdown, so we maybe should strip/validate it before we save/federate it instead of strip it only for display?

**Comment by svbergerem** [comment]
#6642 wasn't about the subject of a conversation but about the shortened last message we display in the left sidebar (inbox). Subjects don't support markdown so we don't have to change anything there. We just don't render it.

Messages however support markdown so when displaying the shortened last message we will have to strip it.

**Comment by SaiVardhan** [comment]
@jhass,

so you don't want the subject to be rendered in markdown and the message should be stripped. right?
`#strip` you mean to remove or show them in markdown content

**Comment by jhass** [comment]
I'd go for `MessageRenderer#plain_text_without_markdown` (notice the `truncate` option) or `MessageRenderer#title`, depending on what looks better.

**Comment by SaiVardhan** [comment]
@jhass 

But ` MessageRenderer#plain_text_without_markdown ` and `MessageRenderer#title` is not working on Message

**Comment by jhass** [comment]
What do you mean not working? Note you need to call them on the return value of `Message#message`, not on `Message#text` nor `Message` itself.

File: app/views/conversations/_conversation.haml [hunk 1]
```diff
@@ -32,7 +32,7 @@
         .last_message
           - if conversation.messages.present?
             %em
-              = conversation.messages.last.text
+              = Diaspora::MessageRenderer.new(conversation.messages.last.text).markdownified
```
Comment (SaiVardhan) [inline app/views/conversations/_conversation.haml]: `Conversation.find(2).messages.last.plain_text_without_markdown` if I use this it is throwing error.
and if I use ` Diaspora::MessageRenderer.new(Conversation.find(2).messages.last).plain_text_without_markdown` it is returning `#<Message:0x0000000ae7d3d0>`

Comment (jhass) [inline app/views/conversations/_conversation.haml]: `Conversation.find(2).messages.last.message.plain_text_without_markdown`.

To explain both of your mistakes:

* `conversation = Conversation.find(2) #=> <Conversation ...>` 
* `messages  = conversation.messages # => [<Message ....>, <Message ...>, ...]` (well, `AR::Relation` really, but conceptually.
* `message = messages.last # => <Message ...> `, note *not* `Diaspora::MessageRenderer`, `Message` does not respond to `plain_text_without_markdown`, only `Diaspora::MessageRenderer` does.
* `message.text #=> "foobar..."`
* `renderer = message.message # => <Diaspora::MessageRenderer ....>` same as `Diaspora::MessageRenderer.new(message.text)`
* Your second snippet effectively did `Diaspora::MessageRenderer.new(message)` and `Message#to_s` is `"<Message ....>"`.

File: spec/controllers/conversations_controller_spec.rb [hunk 2]
```diff
@@ -102,6 +102,14 @@
       get :index, conversation_id: @conversations.first.id
       expect(assigns[:conversation]).to be_nil
     end
+
+    it "retrieves a conversation message with out markdown content " do
+      get :index
+      @conversation = @conversations.first
+      expect(response).to be_success
+      expect(response.body).to match(/cool stuff/)
+      expect(response.body).not_to match(/<strong>cool stuff<\/strong>/)
```
Comment (diaspora-code-review) [inline spec/controllers/conversations_controller_spec.rb]: Use `%r` around regular expression.

File: spec/controllers/conversations_controller_spec.rb [hunk 3]
```diff
@@ -102,6 +102,14 @@
       get :index, conversation_id: @conversations.first.id
       expect(assigns[:conversation]).to be_nil
     end
+
+    it "retrieves a conversation message with out markdown content " do
+      get :index
+      @conversation = @conversations.first
+      expect(response).to be_success
+      expect(response.body).to match(%r{cool stuff})
```
Comment (diaspora-code-review) [inline spec/controllers/conversations_controller_spec.rb]: Use `//` around regular expression.

File: spec/controllers/conversations_controller_spec.rb [hunk 4]
```diff
@@ -102,6 +102,14 @@
       get :index, conversation_id: @conversations.first.id
       expect(assigns[:conversation]).to be_nil
     end
+
+    it "retrieves a conversation message with out markdown content " do
+      get :index
+      @conversation = @conversations.first
+      expect(response).to be_success
+      expect(response.body).to match(%r/cool stuff/)
```
Comment (diaspora-code-review) [inline spec/controllers/conversations_controller_spec.rb]: `%r`-literals should be delimited by `{` and `}`.

File: spec/controllers/conversations_controller_spec.rb [hunk 5]
```diff
@@ -102,6 +102,14 @@
...
+      expect(response.body).not_to match(%r/<strong>cool stuff<\/strong>/)
```
Comment (diaspora-code-review) [inline spec/controllers/conversations_controller_spec.rb]: `%r`-literals should be delimited by `{` and `}`.

**Comment by denschub** [comment]
Just saw the messages by the review bot and they might require some clarification.

Valid regex syntaxes are `%r{foo}` and `/foo/`. The style bot suggested using different syntaxes to avoid escaping. usually, `/foo/` is our preferred form, but if we can avoid having to escape a slash, `%{foo}` is fine, too. If you are looking for `foo/bar`, you'd have to escape the slash in `/foo\/bar/`, but not in `%{foo/bar}`, so the latter is to be preferred.

Feel free to ping someone if you got confused!

File: spec/controllers/conversations_controller_spec.rb (hunk 4 above)
Comment (denschub) [inline spec/controllers/conversations_controller_spec.rb]: See my comment. This should be `/cool stuff/`.

File: spec/controllers/conversations_controller_spec.rb (hunk 5 above)
Comment (denschub) [inline spec/controllers/conversations_controller_spec.rb]: See my comment. This should be `%r{<strong>cool stuff</strong>}`. (Note the unescaped slash!)

**Comment by SaiVardhan** [comment]
A little bit confused on what is preferred form for specifying Regex.

Please specify.

**Comment by jhass** [comment]
You can run the linter locally with `bin/pronto run -c develop`, see https://wiki.diasporafoundation.org/Styleguide#Automatic_local_review Just apply the specific suggestions until it's happy.

You can ignore the cucumber failures, unfortunately we have some that randomly fail on CI.

File: spec/controllers/conversations_controller_spec.rb (hunk 3 above)
Comment (SaiVardhan) [inline spec/controllers/conversations_controller_spec.rb]: Please refer [Rubocop](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/RegexpLiteral) as it suggest to use either `%r{}` or `//`

Comment (jhass) [inline spec/controllers/conversations_controller_spec.rb]: The rule is to use `//` unless the regex contains a `/`. Just follow the suggestions the review bot makes, it's not broken.

**Comment by jhass** [comment]
Merged as 18dbf02181a93af011c74ea6319357bedd3e35ce

Thank you!

**Pull request closed without merging**